strawberryfields/apps/graph/max_clique.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Maximum clique
+==============
+
+**Module name:** :mod:`strawberryfields.apps.graph.max_clique`
+
+.. currentmodule:: strawberryfields.apps.graph.max_clique
+
+This module provides tools for users to identify large cliques in graphs. A clique is a subgraph
+where all nodes are connected to each other. The maximum clique problem is to identify the
+largest clique in a graph. The problem is NP-Hard in a worst-case setting, which is why it is
+valuable to develop better algorithms to tackle it.
+
+The core of this module is a local search algorithm that proceeds as follows:
+
+#. A small clique in the graph is identified. The initial clique can be a single node, or it can
+   be obtained by shrinking a random subgraph, for example obtained from GBS.
+#. The clique is grown as much as possible by adding nodes connected to all nodes in the clique.
+#. When no more growth is possible, a node in the clique is swapped with a node outside of it.
+
+Steps 2-3 are repeated until a pre-determined number of steps have been completed, or until no
+more growth or swaps are possible.
+
+Summary
+-------
+
+.. autosummary::
+    local_search
+
+Code details
+------------
+"""
+import networkx as nx
+
+from strawberryfields.apps.graph import resize
+
+
+def local_search(
+    clique: list, graph: nx.Graph, iterations: int = 100, node_select: str = "uniform"

```

Comment:
what is the motivation for the `iterations = 100` default?

Reply:
I also renamed `iterations` from `steps`, what do you think?

Reply:
Changing `steps` to `iterations` is ok with me. The motivation for the default value is that it is the order of magnitude that made most sense: `10` was too little and `1000` probably too much. However, after thinking about it, I believe it may be best to not have a default at all: the optimum number of iterations will depend on the specific problem at hand and it's probably best for users to give some thought/experiment with this choice.